lua/opencode/context.lua

```diff
@@ -1,53 +1,80 @@
 -- Gathers editor context
 
 local util = require('opencode.util')
-local config = require('opencode.config')
+local config = require('opencode.config') --[[@as OpencodeConfig]]
```

Comment:
Rather than setting the type, maybe we should change the bottom of config.lua to be:

```lua
return setmetatable(M, {
  __index = function(_, key)
    return M.values[key]
  end,
  __newindex = function(_, key, value)
    M.values[key] = value
  end,
  __tostring = function(_)
    return vim.inspect(M.values)
  end,
}) --[[@as OpencodeConfig]]
```